pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -264,8 +264,9 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
     requestContext.setRequestId(requestId);
     if (httpHeaders != null) {
       requestContext.setRequestHttpHeaders(httpHeaders.getRequestHeaders().entrySet().stream().filter(
-              entry -> PinotBrokerQueryEventListenerFactory.getAllowlistQueryRequestHeaders().contains(entry.getKey()))
-          .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+              entry -> PinotBrokerQueryEventListenerFactory.getAllowlistQueryRequestHeaders()
```

Comment:
discussed offline: let's define the semantics in a request context getter/setter.